docs/book.md

```diff
@@ -0,0 +1,299 @@
+      def createUsersWithListInput(respond: UserResource.createUsersWithListInputResponse.type)(body: IndexedSeq[User]): scala.concurrent.Future[UserResource.createUsersWithListInputResponse]
+      def loginUser(respond: UserResource.loginUserResponse.type)(username: String, password: String): scala.concurrent.Future[UserResource.loginUserResponse]
+      def logoutUser(respond: UserResource.logoutUserResponse.type)(): scala.concurrent.Future[UserResource.logoutUserResponse]
+      def getUserByName(respond: UserResource.getUserByNameResponse.type)(username: String): scala.concurrent.Future[UserResource.getUserByNameResponse]
+      def updateUser(respond: UserResource.updateUserResponse.type)(username: String, body: User): scala.concurrent.Future[UserResource.updateUserResponse]
+      def deleteUser(respond: UserResource.deleteUserResponse.type)(username: String): scala.concurrent.Future[UserResource.deleteUserResponse]
+    }
+    object UserResource {
+      def routes(handler: UserHandler)(implicit mat: akka.stream.Materializer): Route = {
+        (post & path("v2" / "user") & entity(as[User])) {
+          body => complete(handler.createUser(createUserResponse)(body))
+        } ~ (post & path("v2" / "user" / "createWithArray") & entity(as[IndexedSeq[User]])) {
+          body => complete(handler.createUsersWithArrayInput(createUsersWithArrayInputResponse)(body))
+        } ~ (post & path("v2" / "user" / "createWithList") & entity(as[IndexedSeq[User]])) {
+          body => complete(handler.createUsersWithListInput(createUsersWithListInputResponse)(body))
+        } ~ (get & path("v2" / "user" / "login") & (parameter(Symbol("username").as[String]) & parameter(Symbol("password").as[String])) & discardEntity) {
+          (username, password) => complete(handler.loginUser(loginUserResponse)(username, password))
+        } ~ (get & path("v2" / "user" / "logout") & discardEntity) {
+          complete(handler.logoutUser(logoutUserResponse)())
+        } ~ (get & path("v2" / "user" / Segment) & discardEntity) {
+          username => complete(handler.getUserByName(getUserByNameResponse)(username))
+        } ~ (put & path("v2" / "user" / Segment) & entity(as[User])) {
+          (username, body) => complete(handler.updateUser(updateUserResponse)(username, body))
+        } ~ (delete & path("v2" / "user" / Segment) & discardEntity) {
+          username => complete(handler.deleteUser(deleteUserResponse)(username))
+        }
+      }
+      ...
+    }
+
+As all parameters are provided as arguments to the function stubs in the trait, there's no concern of forgetting to extract a query string parameter, introducing a typo in a form parameter name, or forgetting to close the bytestream for the streaming HTTP Request.
+
+Separation of business logic
+----------------------------
```

Comment:
Are these sections going to be expanded in some subsequent iteration?

Reply:
Yep, just ended up taking a longer break than intended before finishing them up